migration/migration_test.go

```diff
@@ -16,17 +17,19 @@ func TestMigrateIfNeeded(t *testing.T) {
 	})
 
 	t.Run("Migration should happen on empty DB", func(t *testing.T) {
-		require.NoError(t, migration.MigrateIfNeeded(testDB, utils.MAINNET, utils.NewNopZapLogger()))
+		require.NoError(t, migration.MigrateIfNeeded(context.Background(), testDB, utils.MAINNET, utils.NewNopZapLogger()))
 	})
 
-	version, err := migration.SchemaVersion(testDB)
+	var nilIntermediateState []byte
+	meta, err := migration.SchemaMetadata(testDB)
 	require.NoError(t, err)
-	require.NotEqual(t, 0, version)
+	require.NotEqual(t, uint64(0), meta.Version)
+	require.Equal(t, nilIntermediateState, meta.IntermediateState)
```

Comment:
```suggestion
	require.Nil(t, meta.IntermediateState)
```
And then we can remove the `nilIntermediateState` variable.